test/integration/regression_java_virtualpath_test.go

```diff
@@ -0,0 +1,37 @@
+package integration
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/source"
+)
+
+func TestWarCatalogedCorrectlyIfRenamed(t *testing.T) {
+	// install hudson-war@2.2.1 and renames the file to `/hudson.war`
+	// TODO: is this better expressed in syft/pkg/cataloger/java/archive_parser_test.go?
```

Comment:
agreed, I'm not against having this integration test at all, so consider this comment non-blocking, however, testing specific business logic should be pushed as low as possible test-level-wise (unit is preferred for these kinds of assertions). We do have some regression tests at the integration level to test against the specific asset where we found the regression, so keeping this test here is ok.

Reply:
Thanks for the suggestion! I think I'll leave this here for now; it was reported as an end to end regression, and the parser was still finding the correct package, it was just also finding an incorrect extra package, so the end-to-end flow seems a good place of the test.

I'll push a commit to remove the TODO.